src/compliment/core.clj

```diff
@@ -62,7 +62,8 @@
    - :sort-order (either :by-length or :by-name);
    - :plain-candidates - if true, returns plain strings instead of maps;
    - :extra-metadata - set of extra fields to add to the maps;
-   - :sources - list of source keywords to use."
+   - :sources - list of source keywords to use or a predicate used to filter
+                over source definition maps."
```

Comment:
It feels a bit cludgy to make the `:sources` option now take in both a list of keywords or a predicate function operating over source definition maps. What do folks think? Is it cleaner to add a new option? If so, would the options have to work together?